src/raw/mod.rs

```diff
@@ -1,17 +1,59 @@
 use crate::node::*;
-use crossbeam_epoch::{Atomic, Guard, Owned, Shared};
+use crossbeam_epoch::{Atomic, Guard, Owned, Pointer, Shared};
+use std::borrow::Borrow;
 use std::fmt::Debug;
 use std::sync::atomic::Ordering;
 
 #[derive(Debug)]
 pub(crate) struct Table<K, V> {
     bins: Box<[Atomic<BinEntry<K, V>>]>,
+
+    // since a Moved does not contain associated information,
+    // one instance is sufficient and shared across all bins in this table
+    moved: Atomic<BinEntry<K, V>>,
+    // since the information content of moved nodes is the same across
+    // the table, we share this information
+    //
+    // safety: next_table is a valid pointer if it was read as consequence of reading
```

Comment:
I think this argument has to refer to the fact that the user has a reference to _this_ table.

Reply:
Correct, in particular it should refer to _this_ table being loaded _as_ `HashMap.table`